packages/retail-ui-extensions/src/extension-api/CartApi/CartApiContent.ts

```diff
@@ -0,0 +1,102 @@
+import type {Address, Cart, Customer, CustomSale} from '../Models/Cart';
+
+export type DiscountType = 'Percentage' | 'FixedAmount' | 'Code';
+
+export interface CartApiContent {
+  /** Initial value of the Cart when the screen is rendered. */
+  initialValue: Cart;
+
+  /** Callback when the cart changes */
+  setOnChange(onChange: (cart: Cart) => void): void;
+
+  /** Apply a cart level discount
+   * @param type the type of discount applied (example: 'Percentage')
+   * @param title the title attributed with the discount
+   * @param amount the percentage or fixed monetary amount deducted with the discout
+   */
+  applyCartDiscount(type: DiscountType, title: string, amount: string): void;
+
+  /** Remove the cart discount */
+  removeCartDiscount(): void;
+
+  /** Clear the cart */
+  clearCart(): void;
+
+  /** Set the customer in the cart
+   * @param customer the customer object to add to the cart
+   */
+  setCustomer(customer: Customer): void;
+
+  /** Remove the current customer from the cart */
+  removeCustomer(): void;
+
+  /** Add a custom sale to the cart
+   * @param customSale the custom sale object to add to the cart
+   */
+  addCustomSale(customSale: CustomSale): void;
+
+  /** Add a line item by variant ID to the cart
+   * @param variantId the product variant's numeric ID to add to the cart
+   * @param quantity the number of this variant to add to the cart
+   */
+  addLineItem(variantId: number, quantity: number): void;
+
+  /** Remove the line item at this index from the cart
+   * @param index the index of the line item that should be removed
+   */
+  removeLineItem(index: number): void;
+
+  /** Adds custom properties to the cart
+   * @param properties the custom key to value object to attribute to the cart
+   */
+  addCartProperties(properties: Record<string, string>): void;
+
+  /** Removes the specified cart properties
+   * @param keys the collection of keys to be removed from the cart properties
+   */
+  removeCartProperties(keys: string[]): void;
```

Comment:
These will all technically return promises at runtime, so I'd recommend making that clear in their return types.